internal/controllers/device_data_controller.go

```diff
@@ -276,3 +280,194 @@ func (udc *UserDevicesController) ClearUserDeviceErrorCodeQuery(c *fiber.Ctx) er
 		ClearedAt:  &errCodeQuery.ClearedAt.Time,
 	})
 }
+
+// GetUserDeviceErrorCodeQueriesByTokenID godoc
+// @Summary List all error code queries made for this vehicle.
+// @Tags        error-codes
+// @Param       tokenID path int true "vehicle token id"
+// @Success     200 {object} controllers.GetUserDeviceErrorCodeQueriesResponse
+// @Failure     404 {object} helpers.ErrorRes "Vehicle not found"
+// @Security    BearerAuth
+// @Router      /user/vehicle/{tokenID}/error-codes [get]
+func (udc *UserDevicesController) GetUserDeviceErrorCodeQueriesByTokenID(c *fiber.Ctx) error {
+	tis := c.Params("tokenID")
+	ti, ok := new(big.Int).SetString(tis, 10)
+	if !ok {
+		return fiber.NewError(fiber.StatusBadRequest, fmt.Sprintf("Couldn't parse token id %q.", tis))
+	}
+	tid := types.NewNullDecimal(new(decimal.Big).SetBigMantScale(ti, 0))
+	logger := helpers.GetLogger(c, udc.log)
+
+	userDevice, err := models.UserDevices(
+		models.UserDeviceWhere.TokenID.EQ(tid),
+		qm.Load(models.UserDeviceRels.ErrorCodeQueries, qm.OrderBy(models.ErrorCodeQueryColumns.CreatedAt+" DESC")),
+	).One(c.Context(), udc.DBS().Reader)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return fiber.NewError(fiber.StatusNotFound, "Could not find user device")
+		}
+		logger.Err(err).Msg("error occurred when fetching error codes for device")
+		return fiber.NewError(fiber.StatusInternalServerError, "error occurred fetching device error queries")
+	}
+
+	queries := []GetUserDeviceErrorCodeQueriesResponseItem{}
+
+	for _, erc := range userDevice.R.ErrorCodeQueries {
+		var ercJSON []services.ErrorCodesResponse
+		if err := erc.CodesQueryResponse.Unmarshal(&ercJSON); err != nil {
+			return err
+		}
+
+		userDeviceresp := GetUserDeviceErrorCodeQueriesResponseItem{
+			ErrorCodes:  ercJSON,
+			RequestedAt: erc.CreatedAt,
+			ClearedAt:   erc.ClearedAt.Ptr(),
+		}
+
+		queries = append(queries, userDeviceresp)
+	}
+
+	return c.JSON(GetUserDeviceErrorCodeQueriesResponse{Queries: queries})
+}
+
+// QueryDeviceErrorCodesByTokenID godoc
+// @Summary     Obtain, store, and return descriptions for a list of error codes from this vehicle.
+// @Tags        error-codes
+// @Param       tokenID path int true "vehicle token id"
+// @Param       queryDeviceErrorCodes body controllers.QueryDeviceErrorCodesReq true "error codes"
+// @Success     200 {object} controllers.QueryDeviceErrorCodesResponse
+// @Failure     404 {object} helpers.ErrorRes "Vehicle not found"
+// @Security    BearerAuth
+// @Router      /user/vehicle/{tokenID}/error-codes [post]
+func (udc *UserDevicesController) QueryDeviceErrorCodesByTokenID(c *fiber.Ctx) error {
```

Comment:
Is this the same function as QueryDeviceErrorCodes other than the ud lookup? If so we can just use a function instead of duplicating the code.

Reply:
These are new ones using vehicle tokenID instead of legacy device ID, the old ones will be removed in future releases, we'll also drop the `user_device_id` column from `error_code_queries` table at that point.

Reply:
Let's mark those as deprecated then

Reply:
Added deprecation info